packages/core/src/service/helpers/shipping-calculator/shipping-calculator.ts

```diff
@@ -14,21 +23,32 @@ export class ShippingCalculator {
      * Returns an array of each eligible ShippingMethod for the given Order and sorts them by
      * price, with the cheapest first.
      */
-    async getEligibleShippingMethods(
-        ctx: RequestContext,
-        order: Order,
-    ): Promise<Array<{ method: ShippingMethod; result: ShippingCalculationResult }>> {
+    async getEligibleShippingMethods(ctx: RequestContext, order: Order): Promise<EligibleShippingMethod[]> {
         const shippingMethods = this.shippingMethodService.getActiveShippingMethods(ctx.channel);
-        const eligibleMethods: Array<{ method: ShippingMethod; result: ShippingCalculationResult }> = [];
-        for (const method of shippingMethods) {
-            const eligible = await method.test(order);
-            if (eligible) {
-                const result = await method.apply(order);
-                if (result) {
-                    eligibleMethods.push({ method, result });
-                }
+
+        const checkEligibilityPromises = shippingMethods.map((method) =>
+            this.checkEligibilityByShippingMethod(order, method),
+        );
+        const eligibleMethods = await Promise.all(checkEligibilityPromises);
+
+        // Workaround to remove undefined from type: https://stackoverflow.com/a/51577579
+        const filteredEligibleMethods = eligibleMethods.filter(
```

Comment:
You can use the `notNullOrUndefined` util function here (https://github.com/vendure-ecommerce/vendure/blob/master/packages/common/src/shared-utils.ts#L5)

You can import it from "@vendure/common/lib/shared-utils"

Reply:
Great! When I wrote this part using filter the type definitions didn't get right. Good to know that have a function to deal with this 😁